src/decycle.ts

```diff
@@ -1,36 +1,38 @@
 import { isPlainObject } from './isPlainObject';
 import { pathToPointer } from './pathToPointer';
 
-export const decycle = (obj: unknown, replacer?: (value: any) => any) => {
+export function decycle(obj: unknown, replacer?: (value: any) => any) {
   const objs = new WeakMap<object, string>();
-  return (function derez(value: any, path: string[]) {
-    // The new object or array
-    let curObj: any;
-
-    // If a replacer function was provided, then call it to get a replacement value.
-    if (replacer) value = replacer(value);
+  function derez(value: any, path: (string | number)[]): any {
+    if (replacer) {
+      value = replacer(value);
+    }
 
     if (isPlainObject(value) || Array.isArray(value)) {
-      // The path of an earlier occurance of value
       const oldPath = objs.get(value);
-      // If the value is an object or array, look to see if we have already
-      // encountered it. If so, return a {"$ref":PATH} object.
-      if (oldPath) return { $ref: oldPath };
+
+      if (oldPath) {
+        return { $ref: oldPath };
+      }
```

Comment:
Is there any reason to remove all the helpful comments?

Reply:
Ok I will add this back